File: src/models/session-access-kind.ts
export const SessionAccessKind = {
  ACCOUNT: 0,
  ACCOUNT_CONNECTION: 1,
  DIRECT_CONNECTION: 2,
  TOKEN_ACCOUNT_CONNECTION: 3,
  TOKEN_DIRECT_CONNECTION: 4,
  COOKIE_CONNECTION: 5
} as const;

export type SessionAccessKind = typeof SessionAccessKind[keyof typeof SessionAccessKind];

File: library/src/commonMain/kotlin/models/SessionAccessKind.kt
package models

enum class SessionAccessKind {
    ACCOUNT,
    ACCOUNT_CONNECTION,
    DIRECT_CONNECTION,
    TOKEN_ACCOUNT_CONNECTION,
    TOKEN_DIRECT_CONNECTION,
    COOKIE_CONNECTION;

    companion object {
        fun fromInt(ordinal: Int) = entries.first {it.ordinal == ordinal}
    }
}